feat: Implement Submodel Service Manager

CDiezRodriguez opened the pull request:
## WHAT

- Implemented the `SubmodelServiceManager`.
- Added a new configurable entry in `configuration.yaml` to allow setting a custom root path.
- Updated the Docker image to create a new path for storing files.

## WHY

This implementation is required to enable submodel-related functionality.

## FURTHER NOTES

Closes #186

matbmoser commented on ichub-backend/managers/enablement_services/submodel_service_manager.py:
```diff
@@ -22,40 +22,42 @@
         """Upload a submodel to the service."""
         # Implementation for uploading a submodel
+        if not self.file_system.exists(semantic_id):
+            print("No hay")
```
please remove this

matbmoser reviewed (approved):
remove the line and merge

sm29105 reviewed (approved):
Looks good - please ensure to test if we need to somehow escape the semantic id or if we can put it directly as a folder in the FS. Does it also work on Windows??
If not we could use tools.submodel_type_util to split the semantic id into components.

CDiezRodriguez commented:
> Looks good - please ensure to test if we need to somehow escape the semantic id or if we can put it directly as a folder in the FS. Does it also work on Windows?? If not we could use tools.submodel_type_util to split the semantic id into components.
@sm29105 

- I tested it successfully using the full semantic ID,creating a new one and then reading it back:
![image](https://github.com/user-attachments/assets/1e6b18bd-82c9-456a-b411-423ca70a0a0c)
![image](https://github.com/user-attachments/assets/6e84c5aa-c3c4-4beb-ab34-c607739784d8)

- I haven't tested it on Windows—could you check that?

CDiezRodriguez commented:
I added the SHA-256 hashing as we discussed in the weekly meeting.

CDiezRodriguez commented:
Hi @sm29105, I see no issue with storing the submodels as `{global-id}/{semantic-id-hash}.json` instead of the current structure `{semantic-id-hash}/{global-id}.json`. Is this what you were referring to?

matbmoser commented:
> Hi @sm29105, I see no issue with storing the submodels as `{global-id}/{semantic-id-hash}.json` instead of the current structure `{semantic-id-hash}/{global-id}.json`. Is this what you were referring to?

guys no, store by semantic hash and the global id is the file name

matbmoser reviewed (changes requested):
please change to semantic id hash as folder and global id (uuid) as file. thx :) Can you make sure is a uuid? otherwise it shall fail.

CDiezRodriguez commented:
> please change to semantic id hash as folder and global id (uuid) as file. thx :) Can you make sure is a uuid? otherwise it shall fail.

Currently, the semantic ID hash is used as the folder, and I will ensure that only UUIDs are accepted for the global_id variable.

CDiezRodriguez commented on ichub-backend/config/configuration.yml:
```diff
@@ -28,3 +28,5 @@ authorization:
 database:
   connectionString: "postgresql://user:$DATABASE_PASSWORD@localhost:5432/mydatabase"
   echo: true
+submodel-service:
```
Maybe we should follow Python naming conventions. We already have configuration variables that use this format (https://github.com/eclipse-tractusx/industry-core-hub/blob/main/ichub-backend/config/logging.yml#L24), and I believe they can’t be changed, since they're used for logging configuration in that format. What do you think?

CDiezRodriguez replied:
@matbmoser No, it's more about having a standardized version.

CDiezRodriguez replied:
@matbmoser, both @gomezbc and I discussed this and decided it’s better to follow the Python PEP 8 standard. So, we will use underscores `_` for configuration variables. For example, this variable will be named `submodel_service`.

sm29105 commented:
> > Hi @sm29105, I see no issue with storing the submodels as `{global-id}/{semantic-id-hash}.json` instead of the current structure `{semantic-id-hash}/{global-id}.json`. Is this what you were referring to?
> 
> guys no, store by semantic hash and the global id is the file name

Hi together. Unfortunately I was not part of the meeting today.
I would like to understand the reason for that hard argumentation. Is it security-related or similar? If yes - what's exactly the issue.
The other approach would leave consistent structuring between the DTR and the Submodel Service (=> submodels are a child of the part/twin) and would allow operators of the solution more straight-forward problem analysis (out experience).
I think the rest is easy to be done - hashing the semantic-id and ensuring the global ID is a UUID.

sm29105 reviewed (approved):
Approved from my side
Remark: in "my" PRs I added service specific excpetion class directly in the respective service module. Here we moved the custom exception to a generic place.
We should align on ONE approach in the next days and follow this consequently.

CDiezRodriguez reviewed (commented):
I detected an issue while using the new Manager with the Kubernetes configuration. I'm working on a fix.

CDiezRodriguez commented:
> Approved from my side Remark: in "my" PRs I added service specific excpetion class directly in the respective service module. Here we moved the custom exception to a generic place. We should align on ONE approach in the next days and follow this consequently.

@sm29105 Maybe we can discuss this during the Architecture meeting, or create an issue to refactor the backend code after the main release. That way, we can agree on how we want to structure the code going forward.

CDiezRodriguez commented:
> I detected an issue while using the new Manager with the Kubernetes configuration. I'm working on a fix.

Solved

matbmoser reviewed (approved):
we can merge

CDiezRodriguez commented on ichub-backend/managers/enablement_services/submodel_service_manager.py:
```diff
@@ -22,40 +22,43 @@
     def __init__(self):
-        # TODO: add needed params here
-        pass
+        submodel_service_path = ConfigManager.get_config("submodel-service.path", default={})
```
@matbmoser It depends on the configuration variable.
